src/spac/spatial_analysis.py

```diff
@@ -838,3 +839,109 @@ def _neighborhood_profile_core(
         neighborhood_array = neighborhood_array / bins_areas[np.newaxis, np.newaxis, :]
 
     return neighborhood_array
+
+
+def calculate_spatial_distance(
+    adata,
+    annotation,
+    obsm_key='spatial',
+    subset=None,
+    imageid='imageid',

```

Comment:
Hi @georgezakinih thank you for the comment. I have tested by setting imageid=None but got a KeyError, and then dive deep the scimap sm.tl.spatial_distance function. https://scimap.xyz/Functions/tl/spatial_distance/
Here is the results:
From reviewing the provided scimap function spatial_distance, we can see:
1, Default Parameter: The function signature sets a default for imageid as 'imageid': imageid='imageid';
2. The code then uses imageid to partition the adata object: L123-126
if subset is not None:
    adata_list = [adata[adata.obs[imageid].isin(subset)]]
else:
    adata_list = [adata[adata.obs[imageid] == i] for i in adata.obs[imageid].unique()]
In both cases, it directly references adata.obs[imageid]. This implies that imageid must be a valid column name in adata.obs.
3. If I call spatial_distance with imageid=None, the code will try to evaluate: adata.obs[None], which will raise a KeyError: None since None is not a valid column name. The code never checks if imageid is None; it assumes imageid is always a valid column name.
4. However, the docstring states imageid is optional. Typically, this would mean one of two interpretations:
Interpretation A: You may omit imageid when calling the function, relying on the default 'imageid' column to exist in adata.obs.
Interpretation B: You can pass imageid=None to indicate there's no image-based grouping. However, the code does not currently support this scenario.
Given how the code is written, it aligns with Interpretation A: if you do not specify imageid, it uses the default 'imageid' column name. If your adata does not have such a column, it will fail. It does not handle the case imageid=None gracefully.

Potential solution: refactor calculate_spatial_distance function to handle the case where imageid can be None or a user-defined column name. When imageid is None, the code will not partition by image and will call the scimap function without image stratification by temporarily creating a dummy image column internally.

1. imageid=None Handling:
If imageid is None, the code creates a temporary column (_dummy_imageid) in adata.obs with a single value 'dummy_image' for all cells.
It passes this imageid to scimap so it doesn't raise an error. After computation, it removes this dummy column.

2. User-Defined imageid:
The code now respects any user-defined column name for imageid.
If subset is provided, it checks that all subset labels exist in the provided imageid column.

3. No Partitioning by Image:
When imageid=None, no partitioning is done by image. scimap runs as if there's only one "image" for all cells, since they all share the dummy image ID.